force-app/main/default/classes/WeatherAPICalloutMock.cls

```diff
@@ -0,0 +1,27 @@
+global class WeatherAPICalloutMock implements HttpCalloutMock  {
+
+    private static WeatherService.WeatherData weather;
+
+    public WeatherAPICalloutMock(String location) {
+        if (location == null || location == '') {
+            weather = null;
+        }else { 
+            weather = new WeatherService.WeatherData();
+            weather.temp = 25.5;
+            weather.windSpeed = 3.3;
+            weather.humidity = 75.0;
```

Comment:
I would implement it by the way that constructor accepts WeatherService.WeatherData instance and then assign it to weather variable from 3th line, thus it will be more dynamic without hardcoding